### DISCUSSION TITLE
Livewire @js blade directive still active in Laravel 8.71.0+

### DISCUSSION THREAD

**Discussion by lupinitylabs** [livewire/livewire#4875] [7. Bugs (archived)] [answered]
As outlined in https://github.com/laravel/framework/pull/39522, the @js Blade directive introduced by Livewire seems to take precedence over the native framework implementation, and I am not sure whether this is the intended behavior.

### How to reproduce

In a native Laravel >= 8.71.0 project, `@js(collect(['Hello', 'Caleb']))` will output `JSON.parse('[\u0022Hello\u0022,\u0022Caleb\u0022]')`.

When adding Livewire to the project,the same line will output `JSON.parse(atob('WyJIZWxsbyIsIkNhbGViIl0='));` instead.

Removing the Livewire package, it returns back to the first output.

### Expected behavior

According to above pull request, Livewire should not register its own @js implementation on framework versions >= 8.71.0.
Unless it escaped my attention that the behavior has been changed and a check has been added to the Laravel framework instead, I would expect the output to stay the same in the above scenario.

### Versions checked

The described behavior is consistent in `Laravel versions 8.81.0 and 9.7.0` and was checked with the latest `Livewire version 2.10.5`.

**Comment by calebporzio** [comment, marked as answer]
Thanks for finding this, you're right that Livewire overwrites Laravel's implementation. I would happily accept a PR that checks to see that a "js" directive exists before Livewire registers it's own.

The file and line in question is: src/LivewireServiceProvider.php, line 284

Thanks!

**Comment by lupinitylabs** [comment]
Opened a pull request at https://github.com/livewire/livewire/pull/4876. Let me know if you want anything changed.

**Comment by cthiele-mogic** [comment]
Why this issue is still open?  @js is overwritten by the livewire version, in my case this breaks the application part which does not use livewire....

**Reply by joshhanley** [reply]
@cthiele-mogic this has been fixed by @lupinitylabs in PR #4876 and released today in 2.10.7 🙂
